extensions/common/vault/vault-hashicorp/src/test/java/org/eclipse/edc/vault/hashicorp/HashicorpVaultIntegrationTest.java

```diff
@@ -9,100 +9,281 @@
  *
  *  Contributors:
  *       Mercedes-Benz Tech Innovation GmbH - Initial Test
+ *       Bayerische Motoren Werke Aktiengesellschaft - Refactoring
  *
  */
 
 package org.eclipse.edc.vault.hashicorp;
 
+import org.eclipse.edc.junit.annotations.ComponentTest;
 import org.eclipse.edc.junit.extensions.EdcExtension;
+import org.eclipse.edc.spi.security.CertificateResolver;
 import org.eclipse.edc.spi.security.Vault;
-import org.eclipse.edc.vault.hashicorp.annotations.HashicorpVaultTest;
+import org.eclipse.edc.spi.system.ServiceExtension;
+import org.eclipse.edc.spi.system.ServiceExtensionContext;
+import org.eclipse.edc.spi.system.health.HealthCheckResult;
+import org.eclipse.edc.spi.system.health.HealthCheckService;
+import org.eclipse.edc.spi.system.health.HealthStatus;
+import org.eclipse.edc.spi.system.health.LivenessProvider;
+import org.eclipse.edc.spi.system.health.ReadinessProvider;
+import org.eclipse.edc.spi.system.health.StartupStatusProvider;
+import org.junit.ClassRule;
 import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.DisplayName;
 import org.junit.jupiter.api.Test;
 import org.junit.jupiter.api.extension.ExtendWith;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.ValueSource;
+import org.testcontainers.junit.jupiter.Container;
+import org.testcontainers.junit.jupiter.Testcontainers;
+import org.testcontainers.shaded.org.bouncycastle.operator.OperatorCreationException;
+import org.testcontainers.vault.VaultContainer;
 
+import java.io.IOException;
+import java.security.NoSuchAlgorithmException;
+import java.security.cert.CertificateException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
 import java.util.Map;
 import java.util.UUID;
+import java.util.stream.Collectors;
 
+import static java.lang.String.format;
 import static org.assertj.core.api.Assertions.assertThat;
-import static org.eclipse.edc.vault.hashicorp.HashicorpVaultExtension.VAULT_TOKEN;
-import static org.eclipse.edc.vault.hashicorp.HashicorpVaultExtension.VAULT_URL;
+import static org.eclipse.edc.vault.hashicorp.HashicorpVaultClient.VAULT_DATA_ENTRY_NAME;
+import static org.eclipse.edc.vault.hashicorp.model.Constants.VAULT_TOKEN;
+import static org.eclipse.edc.vault.hashicorp.model.Constants.VAULT_URL;
+import static org.eclipse.edc.vault.hashicorp.util.X509CertificateTestUtil.convertToPem;
+import static org.eclipse.edc.vault.hashicorp.util.X509CertificateTestUtil.generateCertificate;
 
-@HashicorpVaultTest
+@ComponentTest
```

Comment:
maybe I missed this, so from now on all the edge tests with external services will become component tests?
I'm fine with it, but the "component tests" job already takes like 11 minutes, my fear is that adding other tests that involve test containers this time could raise significantly

Reply:
as far as I understand it, testcontainers are spinning up some sort of a proxy container, which then launches the actual Postgres, etc. So running subsequent tests should be pretty quick, at least I could see that on my maching.
I used the `@ComponentTest` annotation, because now `@Postgres....Test` should be reserved for those that actually use an "external" postgres DB, i.e. E2E tests.

If we see that this causes grievances down the line, it should be easy to change.